docs/CLI.md

```diff
@@ -372,7 +372,39 @@ Alias: `-i`. Run all tests serially in the current process, rather than creating
 
 ### `--runTestsByPath`
 
-Run only the tests that were specified with their exact paths.
+Run only the tests that were specified with their exact paths. This avoids converting them into a regular expression and matching it against every single file.
+
+For example, given the following file structure:
+
+```bash
+__tests__
+└── t1.test.js # test
+└── t2.test.js # test
+```
+
+When ran with a pattern, no test is found:
+
+```bash
+jest --runTestsByPath __tests__/t
```

Comment:
Perhaps to is worth mentioning that glob patterns can be used, if users shell is able to expand them?

For example, on Linux or macOS it is possible to run all tests in a folder like this:

```bash
jest --runTestsByPath __tests__/*
```

Reply:
No problem 👍
I used the [test case](https://github.com/facebook/jest/blob/main/e2e/__tests__/runTestsByPath.test.ts) introduced by the original commit as a guideline.